### PR TITLE
ocaml-bos: new port 0.2.1

### PR DESCRIPTION
**Pull request by pguyot**
###### Tested on

macOS 14.4.1 23E224 x86_64
Xcode 15.3 15E204a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?

### CODE REVIEW

File: ocaml/ocaml-bos/Portfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                    interact with the file system and run command line programs.
+
+homepage            https://erratique.ch/software/bos
+master_sites        https://erratique.ch/software/bos/releases/
+
+distname            bos-${version}
+extract.suffix      .tbz
+
+checksums           md5     4cf714c3305d81d485352bb3d7ba3631 \
+                    sha1    4c4c246e286642eed5c7ba40cafc20ab6ab1c5ae \
```
Comment (barracuda156) [inline ocaml/ocaml-bos/Portfile:27]: These are not used anymore and not needed in the portfile.

**Comment by barracuda156** [comment]
Something does not work with its dependency, `ocaml-rresult`:
```
--->  Building ocaml-rresult
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_ocaml_ocaml-rresult/ocaml-rresult/work/rresult-0.7.0" && /opt/local/bin/ocaml /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_ocaml_ocaml-rresult/ocaml-rresult/work/rresult-0.7.0/pkg/pkg.ml build 
Cannot find file /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_ocaml_ocaml-rresult/ocaml-rresult/work/rresult-0.7.0/pkg/pkg.ml.
Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_ocaml_ocaml-rresult/ocaml-rresult/work/rresult-0.7.0" && /opt/local/bin/ocaml /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_ocaml_ocaml-rresult/ocaml-rresult/work/rresult-0.7.0/pkg/pkg.ml build 
Exit code: 2
Error: Failed to build ocaml-rresult: command execution failed
```

File: ocaml/ocaml-bos/Portfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+description         Basic OS interaction for OCaml
+long_description    Bos provides support for basic and robust interaction with \
+                    the operating system in OCaml. It has functions to access \
+                    the process environment, parse command line arguments, \
+                    interact with the file system and run command line programs.
+
+homepage            https://erratique.ch/software/bos
+master_sites        https://erratique.ch/software/bos/releases/
+
```
Comment (barracuda156) [inline ocaml/ocaml-bos/Portfile:22]: @pguyot Could you please add `use_bzip2 yes`? Like in https://github.com/macports/macports-ports/pull/23600
That will fix extraction.

Comment (ryandesign) [reply ocaml/ocaml-bos/Portfile:22]: > That will fix extraction.

…on OS X 10.8 and earlier.

**Pull request merged**